### PR TITLE
[FIX] Contextual errors for this and RegExp declarations in IRC module

### PR DESCRIPTION
**Pull request by Pharserror**
@RocketChat/core 

_Should Close_ #6902 and #7412
I say _Should_ because I have not been able to deploy this remotely and have only tested the RC fixes locally.

I was getting several "this.someFunction is undefined" errors after enabling the IRC plugin via the admin panel in Chrome 61. I fixed these context errors by binding `this` to the functions that were giving the `undefined` message in the constructor.
Also, there are several `RegExp`'s that were declared such that the functions they were made to trigger were not being triggered. I changed the declarations and now I am seeing messages.

### CODE REVIEW

File: packages/rocketchat-irc/server/server.js [hunk 1]
```diff
@@ -44,10 +44,13 @@ class IrcClient {
 		this.socket.setNoDelay;
 		this.socket.setEncoding('utf-8');
 		this.socket.setKeepAlive(true);
+		this.connect = this.connect.bind(this);
+		this.onConnect = this.onConnect.bind(this);
```
Comment (graywolf336) [inline packages/rocketchat-irc/server/server.js]: Hmmm unless I'm missing something, this gets overwrote by the next line?

Comment (graywolf336) [reply packages/rocketchat-irc/server/server.js]: Derp, I was missing something. :see_no_evil: so never mind!

Comment (Pharserror) [reply packages/rocketchat-irc/server/server.js]: Yeah the bind there is something else to fire off other events. Maybe that function name should change?

Comment (Pharserror) [reply packages/rocketchat-irc/server/server.js]: 👍

**Comment by Pharserror** [comment]
Not sure if the failing test is because of my changes. Can we run the Travis CI build again?

**Comment by rodrigok** [comment]
@Pharserror we moved to CircleCI, everything passed, I just updated your branch with our develop branch to run the CI again.

**Comment by Pharserror** [comment]
👌

**Pull request merged**